src/Sarif.Converters/AndroidStudioConverter.cs

```diff
@@ -150,51 +152,73 @@ public Result ConvertProblemToSarifResult(AndroidStudioProblem problem)
                     Region = problem.Line <= 0 ? null : Extensions.CreateRegion(problem.Line)
                 };
 
-                RemoveBadRoot(file, out uri);
+                bool foundUriBaseId = RemoveBadRoot(file, out uri);
                 location.PhysicalLocation.FileLocation.Uri = uri;
+
+                // TODO enable this change after current work
+                // https://github.com/Microsoft/sarif-sdk/issues/1168
+                // 
+                //if (foundUriBaseId)
+                //{
+                //    location.PhysicalLocation.FileLocation.UriBaseId = PROJECT_DIR;
+                //}
             }
 
             result.Locations = new List<Location> { location };
 
             return result;
         }
 
-        private string CreateFullyQualifiedLogicalName(AndroidStudioProblem problem)
+        private string CreateFullyQualifiedLogicalName(AndroidStudioProblem problem, out int index)
```

Comment:
>CreateFullyQualifiedLogicalName [](start = 23, length = 31)

This method actually does three things, and although it's hard to come up with a good method name that conveys that [*], it's at least worth a summary comment:
1. It adds an entry to the LogicalLocations array for each nested level of the logical location where the problem occurs.
2. It builds up and returns the fully qualified name of the most deeply nested location.
3. It sets index to the index within LogicalLocations of the newly created entry for the most deeply nested location.

[*] Usually a code smell that the method should be refactored, but let that be for another day.

Reply:
On further reflection, I think a good method name is `AddLogicalLocationAndAncestors`. It returns information about the (most deeply nested) logical location.

---
In reply to: [243866307](https://github.com/Microsoft/sarif-sdk/pull/1170#discussion_r243866307) [](ancestors = 243866307)